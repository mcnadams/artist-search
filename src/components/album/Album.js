import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Album({ album }) {
  const path = `/album/${album.id}`;
  const date = album.date.slice(0, 4);
  return (
    <>
    <Link to={path}>
      <img src={album.image} style={{ width: '200px' }} />
      <p>{album.title} ({date})</p>
    </Link>
    </>
  );
}

Album.propTypes = {
  album: PropTypes.object.isRequired
};

export default Album;
