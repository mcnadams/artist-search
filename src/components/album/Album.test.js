import React from 'react';
import { shallow } from 'enzyme';
import Album from './Album';

describe('Album renders', () => {
  it('renders Album', () => {


    const wrapper = shallow(<Album artist={'lady'} album={{ date: 'yes', id: 'no' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
