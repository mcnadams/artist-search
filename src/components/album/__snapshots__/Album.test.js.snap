// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album renders renders Album 1`] = `
<Fragment>
  <Link
    to="/album/lady/no"
  >
    <img
      style={
        Object {
          "width": "200px",
        }
      }
    />
    <p>
       (
      yes
      )
    </p>
  </Link>
</Fragment>
`;
